chunky/src/java/se/llbit/chunky/renderer/TileBasedRenderer.java

```diff
@@ -57,6 +58,9 @@ public void setPostRender(BooleanSupplier callback) {
     postRender = callback;
   }
 
+  protected Supplier<WorkerState> workerStateFactory = WorkerState::new;
+  protected Supplier<Ray> rayFactory = Ray::new;
```

Comment:
Any arguments against adding PluginAPI to these factory fields?

Reply:
Why do we use the factory pattern here in the first place? Are there plugins that need this? If we add PluginApi, we have to maintain compatibility, which may not be what we want during refactoring.

Reply:
I had a renderer plugin idea which needed a way to add additional worker state / ray state --> answering your question: yes, plugins could need this.
But I would not mind it being internal (allowing for breaking changes) for now instead of PluginAPI.

Reply:
I have a PR with a per-worker pool of rays, which are re-used. Maybe I'll move the ray factory into the worker state eventually.

Reply:
Sounds good, I have no problem with updating plugin code for unstable/snapshot features ^^